contract/src/main.rs

```diff
@@ -843,7 +843,7 @@ pub extern "C" fn set_approval_for_all() {
     for token_id in owned_tokens {
         // We assume a burned token cannot be approved
         if utils::is_token_burned(&token_id) {
```

Comment:
This does not look legit to have a condition to only continue in all cases of the condition, these lines if not effective should be removed. 

A also at the same time it seems strange to manipulate a burnt token. I'll check that with L2.